File: python-solutions/problem010.py
from time import time


def prime_sieve(n):
    size = n//2
    sieve = [1]*size
    limit = int(n**0.5)
    for i in range(1,limit):
        if sieve[i]:
            val = 2*i+1
            tmp = ((size-1) - i)//val 
            sieve[i+val::val] = [0]*tmp
    return [2] + [i*2+1 for i, v in enumerate(sieve) if v and i>0]


def compute():
    return sum(prime_sieve(2000000))

if __name__ == "__main__":
    start = time()
    answer = compute()
    end = time() - start

    print(f"\nAnswer for Problem 10: {answer}")
    print( f"Time Taken: {end} seconds\n")

File: golang-solutions/problem010.go
package main

import (
	"fmt"
	"math"
	"time"
)

func main() {
	start := time.Now()
	var answer int64

	for i := int64(0); i < 2000000; i++ {
		if isPrime(i) {
			answer += i
		}
	}

	fmt.Printf("\nAnswer to Problem 10: %d\n", answer-1)
	fmt.Printf("Time Taken: %v\n\n", time.Since(start))
}

func isPrime(n int64) bool {
	for i := int64(2); i <= int64(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}
